src/app/modules/registration/components/validation/validation.component.ts

```diff
@@ -1,12 +1,29 @@
-import { Component, OnInit, ChangeDetectionStrategy, Input, AfterViewChecked } from '@angular/core';
+import { Component, OnInit, ChangeDetectionStrategy, Input, ChangeDetectorRef, OnDestroy, Output } from '@angular/core';
 import { FormControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+import * as EventEmitter from 'events';
 
 @Component({
   selector: 'app-validation',
   templateUrl: './validation.component.html',
   styleUrls: ['./validation.component.scss'],
-  //changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ValidationComponent {
+export class ValidationComponent implements OnInit, OnDestroy {
+
   @Input() control: FormControl;
+  constructor(private changeDetector: ChangeDetectorRef) { }
+
+  controlSub: Subscription;
+
+  ngOnInit(): void {
+    this.controlSub = this.control.statusChanges.subscribe(() => {
+      this.changeDetector.markForCheck();
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.controlSub.unsubscribe();
```

Comment:
Spróbuj sobie z untilDestroy - https://github.com/ngneat/until-destroy
Jak raz spróbujesz to potem nie zrezygnujesz :)

Reply:
done - uzyte @UntilDestroy({ checkProperties: true })